### PR TITLE
Notion (Independent Publisher Connector)

### PR DESCRIPTION
**Pull request by sekharmalla**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [ ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by Amjed-Ayoub** [comment]
Hello @sekharmalla,
Thank you very much and congratulations on your update submission. Our Certification Engineers will review it within 1-2 weeks.
Please let us know if you have any questions.
Thank you very much again for working with us.

File: independent-publisher-connectors/Notion/apiDefinition.swagger.json [hunk 1]
```diff
@@ -256,66 +286,131 @@
...
+            "required": true,
+            "type": "string",
+            "x-ms-visibility": "internal",
+            "default": "2022-06-28"
+          }
+        ]
+      },
+      "patch": {
+        "responses": {
+          "default": {
```
Comment (sriyen-msft) [inline independent-publisher-connectors/Notion/apiDefinition.swagger.json:396]: @sekharmalla Default responses are not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide valid expected response codes with schema definitions for all the operations in the swagger. **This applies to all the default responses in the swagger.**

File: independent-publisher-connectors/Notion/apiDefinition.swagger.json [hunk 2]
```diff
@@ -256,66 +286,131 @@
...
+            "description": "default",
+            "schema": {}
```
Comment (sriyen-msft) [inline independent-publisher-connectors/Notion/apiDefinition.swagger.json:398]: @sekharmalla Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger. **This applies to all the empty schemas in the swagger.**

File: independent-publisher-connectors/Notion/apiDefinition.swagger.json [hunk 3]
```diff
@@ -330,21 +425,223 @@
...
+                      "description": "rich_text",
+                      "title": "Heading 2"
+                    },
+                    "color": {
+                      "type": "string",
+                      "description": "Color of the block.",
+                      "title": "Color"
+                    }
+                  },
+                  "description": "heading_2",
```
Comment (sriyen-msft) [inline independent-publisher-connectors/Notion/apiDefinition.swagger.json:536]: @sekharmalla All the description values in the swagger are expected to be free from grammatical and spell errors. Please provide valid description values for all the properties.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/Notion/apiDefinition.swagger.json (hunk 1 above)
Comment (sekharmalla) [reply independent-publisher-connectors/Notion/apiDefinition.swagger.json]: Thanks, @sriyen-msft, 

I have updated the Default response with the Respective response codes. and most of the actions in this swagger have dynamic responses hence not defining.

Thanks

File: independent-publisher-connectors/Notion/apiDefinition.swagger.json (hunk 2 above)
Comment (sekharmalla) [reply independent-publisher-connectors/Notion/apiDefinition.swagger.json]: Hi,
most of the actions in this swagger have dynamic responses hence not defining. 

Please advice 
Thanks,

File: independent-publisher-connectors/Notion/apiDefinition.swagger.json (hunk 3 above)
Comment (sekharmalla) [reply independent-publisher-connectors/Notion/apiDefinition.swagger.json]: Updated proper descriptions for the swagger file.

Thanks.

File: independent-publisher-connectors/Notion/apiDefinition.swagger.json [hunk 4]
```diff
@@ -330,21 +424,224 @@
...
+                      "title": "Heading 2"
+                    },
+                    "color": {
+                      "type": "string",
+                      "description": "Color of the block.",
+                      "title": "Color"
+                    }
+                  },
+                  "description": "Heading one block objects contain the Rich
+_text within the heading_1 property",
```
Comment (sriyen-msft) [inline independent-publisher-connectors/Notion/apiDefinition.swagger.json:536]: @sekharmalla It is a syntax error, please remove the new line from the description value.

File: independent-publisher-connectors/Notion/apiDefinition.swagger.json [hunk 5]
```diff
@@ -635,9 +1078,778 @@
...
+                          "type": "string",
+                          "description": "Page cover image URL.",
+                          "title": "Cover URL",
+                          "x-ms-visibility": "important"
+                        }
+                      },
+                      "description": "external"
+                    }
+                  },
+                  "description": "Page cover image.	"
```
Comment (sriyen-msft) [inline independent-publisher-connectors/Notion/apiDefinition.swagger.json:1545]: @sekharmalla Please remove extra spaces from the any value in the swagger.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]
@sekharmalla Please update your private branch to resolve the branch conflicts.

File: independent-publisher-connectors/Notion/apiDefinition.swagger.json (hunk 4 above)
Comment (sekharmalla) [reply independent-publisher-connectors/Notion/apiDefinition.swagger.json]: I have Updated the syntax error. @sriyen-msft Thanks.

File: independent-publisher-connectors/Notion/apiDefinition.swagger.json (hunk 5 above)
Comment (sekharmalla) [reply independent-publisher-connectors/Notion/apiDefinition.swagger.json]: I removed Extra space from the description. I will double-check again. but @sriyen-msft  can you advise me on how do I fix the conflict error? 

I have updated the same changes on my original branch also.

**Comment by Amjed-Ayoub** [comment]
Hello @sekharmalla,
I hope you are doing well.
Congratulations, your connector is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by v-EgorKozhadei** [comment]
Hello @sekharmalla, 

I hope you are doing well. 

Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps. 

Please let us know if you have any questions. 

Thank you very much for working with us.

**Comment by sriyen-msft** [comment]
@sekharmalla Congratulations! The deployment of your connector and documentation page is now complete, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program, Thanks.

**Pull request merged**